File: THCS/2022_2023/HaNoi_2223/Bai1/TG.py
def main():
    # Đọc dữ liệu từ tệp TG.INP
    with open("TG.INP", "r") as input_file:
        N = int(input_file.readline().strip())  # Số lượng thí sinh
        times = [int(input_file.readline().strip()) for _ in range(N)]  # Thời gian hoàn thành của từng thí sinh

    # Thời gian bắt đầu là 8 giờ 00 phút
    start_hour = 8
    start_minute = 0

    # Danh sách để lưu thời gian kết thúc
    end_times = []

    for Ti in times:
        # Tính thời gian kết thúc
        end_minute = start_minute + Ti  # Tính phút kết thúc
        end_hour = start_hour + end_minute // 60  # Tính giờ kết thúc
        end_minute = end_minute % 60  # Tính phút còn lại sau khi chia cho 60

        # Định dạng thời gian
        end_time = f"{end_hour:02}:{end_minute:02}"
        end_times.append(end_time)

    # Ghi kết quả ra tệp TG.OUT
    with open("TG.OUT", "w") as output_file:
        for end_time in end_times:
            output_file.write(end_time + "\n")

if __name__ == "__main__":
    main()

File: THCS/2022_2023/HaNoi_2223/Bai1/TG.cpp
#include <iostream>
#include <fstream>
#include <iomanip>
using namespace std;

int main() {
    ifstream inputFile("TG.INP");
    ofstream outputFile("TG.OUT");

    int N; // Số lượng thí sinh
    inputFile >> N;

    // Thời gian bắt đầu là 8 giờ 00 phút
    int startHour = 8;
    int startMinute = 0;

    for (int i = 0; i < N; ++i) {
        int Ti; // Thời gian hoàn thành bài sát hạch của thí sinh i
        inputFile >> Ti;

        // Tính thời gian kết thúc
        int endMinute = startMinute + Ti; // Tính phút kết thúc
        int endHour = startHour + endMinute / 60; // Tính giờ kết thúc
        endMinute = endMinute % 60; // Tính phút còn lại sau khi chia cho 60

        // Ghi kết quả ra tệp
        outputFile << setfill('0') << setw(2) << endHour << ":" << setfill('0') << setw(2) << endMinute << endl;
    }

    inputFile.close();
    outputFile.close();

    return 0;
}
